superset-frontend/src/SqlLab/components/ResultSet/index.tsx

```diff
@@ -164,11 +164,7 @@ const ResultSet = ({
       setCachedData(query.results.data);
       dispatch(clearQueryResults(query));
     }
-    if (
-      query.resultsKey &&
-      prevQuery?.resultsKey &&
-      query.resultsKey !== prevQuery.resultsKey
-    ) {
+    if (query.resultsKey && query.resultsKey !== prevQuery?.resultsKey) {
```

Comment:
is there any case where the `refetch button` will appear since we are removing this case?

Reply:
I don't think so. Before the refactor the if statement looked like this:
```
if (
      nextProps.query.resultsKey &&
      nextProps.query.resultsKey !== this.props.query.resultsKey
    ) 
```
So I don't think `prevQuery?.resultsKey &&` was supposed to be there in the first place.